Sources/MapboxMaps/Offline/OfflineCallbacks.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+/// Returns a closure suitable for the OfflineManager and TileStore callback based
+/// APIs, that converts the expected type into a Swift Result type.
+/// - Parameters:
+///   - closure: developer provided completion closure.
+///   - type: The ObjC type. For example, for `[TileRegion]` this would be `NSArray`
+/// - Returns: A suitable `MBXExpected` base closure.
+internal func coreAPIClosureAdapter<T, ErrorType, ObjCType>(for closure: @escaping (Result<T, ErrorType>) -> Void, type: ObjCType.Type)
+-> ((MBXExpected<AnyObject, AnyObject>?) -> Void)
+where ObjCType: AnyObject,
+      ErrorType: CoreErrorRepresentable,
+      ErrorType.CoreErrorType: AnyObject {
```

Comment:
```suggestion
internal func coreAPIClosureAdapter<T, ErrorType, ObjCType>(
    for closure: @escaping (Result<T, ErrorType>) -> Void,
    type: ObjCType.Type) -> ((MBXExpected<AnyObject, AnyObject>?) -> Void) where ObjCType: AnyObject,
                                                                                 ErrorType: CoreErrorRepresentable,
                                                                                 ErrorType.CoreErrorType: AnyObject {
```

This wrapping feels a little easier to read to me